pkg/osvscanner/osvscanner.go

```diff
@@ -505,6 +516,15 @@ func parseLockfilePath(lockfileElem string) (string, string) {
 	return splits[0], splits[1]
 }
 
+type Package struct {
```

Comment:
Hmm... now that I think about it, do we even need this intermediate type? It looks like what we're looking for is just the `osv.Query` type, but rather than being appended to a pointer being passed in, we return it as an array. 

This should reduce the number of changes while still accomplishing what you want right?

Reply:
I make a separate Package object because it doesn't make semantic sense to use an osv.Query object to create a depsdev query. While it might be possible to do so, I think it's a misuse of the meaning of the osv.Query object and we should be making osv and depsdev queries from an abstracted package object instead.

In the case where commit information and package information are available, it's valid to just construct an osv.Query that just contains just the commit [1]. in this case we wouldn't be able to construct a depsdev query from the osv query. I'm not sure whether this scenario exists yet, but I think it illustrates why it's cleaner to have a separate package object.

As discussed I'll make this a private type and submit it as is!